service/src/LevelZero.cpp

```diff
@@ -523,6 +526,366 @@ namespace geopm
+//
+//            // Destroy Event Pool
+//            ze_result = zeEventPoolDestroy(m_devices.at(l0_device_idx).event_pool);
+//            check_ze_result(ze_result, GEOPM_ERROR_RUNTIME,
+//                            "LevelZero::" + std::string(__func__) +
+//                            ": LevelZero Metric Event Pool Destroy failed",
+//                            __LINE__);
+//
+//            // Deactivate the device context
+//            ze_context_handle_t context = m_devices.at(l0_device_idx).context;
+//            ze_result = zetContextActivateMetricGroups(context,
+//                                                       m_devices.at(l0_device_idx).device_handle,
+//                                                       0, nullptr);
+//            check_ze_result(ze_result, GEOPM_ERROR_RUNTIME,
+//                            "LevelZero::" + std::string(__func__) +
+//                            ": LevelZero Metric Context Deactivation failed",
+//                            __LINE__);
+//        }
+    }
+
+
+    void LevelZeroImp::metric_init(unsigned int l0_device_idx, unsigned int l0_domain_idx)
+    {
+        ze_result_t ze_result;
+        ze_context_handle_t context = m_devices.at(l0_device_idx).subdevice.context.at(l0_domain_idx);
+        ze_result = zetContextActivateMetricGroups(context, m_devices.at(l0_device_idx).subdevice_handle.at(l0_domain_idx),
+                                                   1, &m_devices.at(l0_device_idx).subdevice.metric_group_handle.at(l0_domain_idx));
+
+        ze_event_pool_handle_t event_pool_handle = nullptr;
+        ze_event_pool_desc_t event_pool_desc = {ZE_STRUCTURE_TYPE_EVENT_POOL_DESC, nullptr, 0, 1};
+
+        ze_result = zeEventPoolCreate(context, &event_pool_desc, 1,
+                                      &m_devices.at(l0_device_idx).subdevice_handle.at(l0_domain_idx),
+                                      &event_pool_handle);
+
+        check_ze_result(ze_result, GEOPM_ERROR_RUNTIME,
+                        "LevelZero::" + std::string(__func__) +
+                        ": LevelZero Event Pool Create failed",
+                        __LINE__);
+        m_devices.at(l0_device_idx).subdevice.event_pool.push_back(event_pool_handle);
+
+        ze_event_desc_t event_desc = {ZE_STRUCTURE_TYPE_EVENT_DESC, nullptr, 0,
+                                      ZE_EVENT_SCOPE_FLAG_HOST, ZE_EVENT_SCOPE_FLAG_HOST};
+
+        ze_event_handle_t event = nullptr;
+        ze_result = zeEventCreate(event_pool_handle, &event_desc, &event);
+        check_ze_result(ze_result, GEOPM_ERROR_RUNTIME,
+                        "LevelZero::" + std::string(__func__) +
+                        ": LevelZero Event Create failed",
+                        __LINE__);
+
+        m_devices.at(l0_device_idx).subdevice.event.push_back(event);
+
+        zet_metric_streamer_desc_t metric_streamer_desc = {
+            ZET_STRUCTURE_TYPE_METRIC_STREAMER_DESC,
+            nullptr,
+            32768,
```

Comment:
Why this number of reports per notification?

Reply:
Updated to 4 reports (1ms per report target)